### PR TITLE
arvo: embrace filesystem, add %lull

### PR DESCRIPTION
**Pull request by joemfb**
This PR further commits to the arvo filesystem added in #4010, using it exclusively for all kernel updates. Depending on the filesystem to this degree further decouples arvo upgrades from %clay. And it lets us update the pill less frequently -- since the kernel will be reset on any source change, it's no longer possible to boot into a mismatch between running kernel and source (without deliberately producing a bad pill through a non-standard process).

It also adds a placeholder %lull module, for kernel-wide structures. The first core in %zuse will be moved into %lull.

### CODE REVIEW

**Comment by joemfb** [comment]
This PR also removes %give to arvo (by way of intercepted %give to the Outside), updating both %clay and %dill to exclusively %pass to arvo.

And one more benefit of this further commitment to the arvo filesystem: kernel updates no longer trigger unnecessary rebuilds of the "reef" (the system-wide subject: zuse/lull/arvo/hoon stack), because %clay scry's into arvo to check for source matches with the running kernel. Previously, it check the latest commit on the %home desk, but this check would fail in the case of a kernel upgrade, because the commit to %home had not yet been finalized.

**Review by belisarius222 — APPROVED** [review]
LGTM

**Review by philipcmonk — APPROVED** [review]
Much cleaner!

File: pkg/arvo/gen/reload-event.hoon [hunk 1]
```diff
@@ -1,26 +0,0 @@
-::  Produce a raw event to reload a vane
```
[the remark this answers is not in the document]
Comment (joemfb) [reply pkg/arvo/gen/reload-event.hoon:1]: You can bypass clay and reload anything in the kernel, but only from source (ie, an `%what` event). And the update will only occur if the source is different from what's running (no more fixpoint `|reset`).

I should add a generator for producing such an event, but i'm not sure what the interface should be -- everything on a desk?

[the remark this answers is not in the document]
Comment (philipcmonk) [reply pkg/arvo/gen/reload-event.hoon:1]: In practice you usually just want to update a single vane, and ideally you'd not change the other ones without having to know exactly what they are.  But if you do that's fine too

[the remark this answers is not in the document]
Comment (joemfb) [reply pkg/arvo/gen/reload-event.hoon:1]: We could always `+cat` a comment with some entropy to force reload ... I'll think about adding some variants in a separate PR (`+file-ovum2` in `lib/pill` is the basic mechanism)..

**Pull request merged**